packages/core/src/App/Containers/Layout/footer.jsx

```diff
@@ -50,6 +52,8 @@ const Footer = ({
         footer_extensions_right = footer_extensions.filter(footer_extension => footer_extension.position === 'right');
     }
 
+    setUrlLanguage(language);
```

Comment:
```suggestion
    React.useEffect(() => {
        setUrlLanguage(language);
    }, [language]);
```
@yauheni-kryzhyk-deriv it's better to set the language when only `language` prop has changed, not every time we re-render the Footer.
But also, Is this line necessary here at all? Did it not work when you just added language variable in connect here? It triggers the component re-render together with its GoToDeriv child, no? 🙂

Reply:
Actually, I agree with you about useEffect. 
But in our case this function just sets the language to local variable without any side effects. I believe it is much cheaper then implement useEffect and observe this variable. Because with every click on the footer it will be rerendered. So useEffect will execute always.
Regarding to your second question: no :) it is not enough, because the function for creating the path takes lang variable from the closure in the same document. This variable is updated only after rerendering the app.
So in this case both approaches don’t work. Because useEffect will trigger this function, but not rerender the component. In this case we need implement something for triggering rerender (f.e. usestate).
but anyway, thanks for your review - it helped me to overthink it, and I believe  I have found better solution and I will recreate a PR from master and implement it there (this one was created from develop, what is incorrect as I've got know)